thread/thread.cpp

```diff
@@ -135,13 +135,15 @@ namespace photon
 #if defined(__linux__)
         madvise(ptr, stack_size, MADV_NOHUGEPAGE);
 #endif
+        mprotect(ptr, PAGE_SIZE, PROT_NONE);
```

Comment:
should we move the default allocator into stack-allocator.cpp ?

Reply:
done